Rust/mechanics_fast/src/lib.rs

```diff
@@ -1,12 +1,28 @@
-use cgmath::{Vector3, InnerSpace};
+use cgmath::{InnerSpace, Vector3};
 
-#[no_mangle]
-pub extern "C" fn compute_gravitational_acceleration(displacement: Vector3<f64>, m2: f64) -> Vector3<f64> {
-    const GRAVITATIONAL_CONSTANT: f64 = 6.67430e-11;
+fn compute_gravitational_acceleration(displacement: Vector3<f32>, mass: f32) -> Vector3<f32> {
+    let g: f32 = 9.81;
 
-    let distance = displacement.magnitude();
+    let distance_squared = displacement.magnitude2();
 
-    let acceleration = displacement * (GRAVITATIONAL_CONSTANT * m2) / (distance * distance * distance);
+    if distance_squared > 0.0 {
+        let magnitude = g * mass / distance_squared.sqrt();
+        displacement.normalize() * magnitude
+    } else {
+        Vector3::new(0.0, 0.0, 0.0)
+    }
+}
```

Comment:
I think the original `compute_gravitational_acceleration` from d9534bdb890232396a9ac93604094880d23bef84 was perfect. The modified one is more complicated and also incorrect; it uses 9.81, which is specifically the gravitational acceleration on Earth's surface, and not applicable anywhere else. It also had the flags that allowed the C# code to call it. For reference, it was:

```
#[no_mangle]
pub extern "C" fn compute_gravitational_acceleration(displacement: Vector3<f64>, m2: f64) -> Vector3<f64> {
    const GRAVITATIONAL_CONSTANT: f64 = 6.67430e-11;

    let distance = displacement.magnitude();

    let acceleration = displacement * (GRAVITATIONAL_CONSTANT * m2) / (distance * distance * distance);

    acceleration
}
```